geode-core/src/main/java/org/apache/geode/internal/cache/execute/AbstractExecution.java

```diff
@@ -504,8 +508,16 @@ private void handleException(Throwable functionException, final Function fn,
         ((InternalResultSender) sender).setException(functionException);
       }
     } else {
-      logger.warn("Exception occurred on local node while executing Function:",
-          functionException);
+      if (AbstractExecution.SUPPRESS_FUNCTION_EXCEPTION_LOGGING
+          && functionException instanceof FunctionException) {
+        if (logger.isDebugEnabled()) {
```

Comment:
Isn't there a way to do this in log4j with markers or something? It seems rather heavy to need an external parameter to toggle the log-level or verbosity of this message.

Reply:
Thanks @pivotal-jbarrett, I agreed with you, log4j with markers approach may be more careful and flexible. I've added a new commit with the changes.
@albertogpz Do you have any objections to it?

Reply:
Using Markers looks like a cleaner solution.